### PR TITLE
SDIT-1623 Add ability to search on multiple template names and add document ids to CSIP Details

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/core/DocumentResource.kt [hunk 1]
```diff
@@ -98,13 +100,19 @@ class DocumentResource(private val documentService: DocumentService) {
   )
   fun getDocumentIds(
     @Schema(description = "The booking id") @PathVariable bookingId: Long,
-    @Schema(description = "The template name") @PathVariable templateName: String,
-  ) = documentService.findAllIds(bookingId, templateName)
+    @RequestParam(value = "templateName", required = true)
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/core/DocumentResource.kt:103]:
```suggestion
    @RequestParam(value = "templateNames", required = true)
```

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/core/DocumentResource.kt [hunk 2]
```diff
@@ -98,13 +100,19 @@ class DocumentResource(private val documentService: DocumentService) {
...
+    @Parameter(
+      description = "The unique name of the template used for a document",
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/core/DocumentResource.kt:105]: think this should now be plural, with multiple examples?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/core/DocumentResource.kt [hunk 3]
```diff
@@ -98,13 +100,19 @@ class DocumentResource(private val documentService: DocumentService) {
...
+    )
+    templateNames: List<String>,
+  ) =
+    documentService.findAllIds(bookingId, templateNames)
 }
 
 @JsonInclude(JsonInclude.Include.NON_NULL)
 @Schema(description = "Document id")
 data class DocumentIdResponse(
-  @Schema(description = "The document id", required = true)
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/core/DocumentResource.kt:108]: was this removed as unnecessary?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/core/DocumentResource.kt (hunk 2 above)
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/core/DocumentResource.kt:105]: this confused me when I first saw it, might be worth adding that this parameter can be repeated.  Would be interested to see what swagger is generated for it.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/core/DocumentResourceIntTest.kt [hunk 4]
```diff
@@ -166,17 +166,25 @@ class DocumentResourceIntTest : IntegrationTestBase() {
...
+      webTestClient.get().uri("/documents/booking/$bookingId")
+        .headers(setAuthorisation(roles = listOf("ROLE_NOMIS_DOCUMENTS")))
+        .exchange()
+        .expectStatus().isBadRequest
+    }
+
     @Test
     fun `will return a document id filtered by bookingId and template name`() {
-      webTestClient.get().uri("/documents/booking/$bookingId/template/CSIPA1_FNP")
+      webTestClient.get().uri("/documents/booking/$bookingId?templateName=CSIPA1_FNP")
```
Comment (petergphillips) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/core/DocumentResourceIntTest.kt:187]: would be nice to have a test where you're passing in multiple `templateName` parameters

**Review by petergphillips — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/core/DocumentResource.kt (hunk 3 above)
Comment (karenmillermoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/core/DocumentResource.kt:108]: yeah - cos it's just the response

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/core/DocumentResource.kt (hunk 2 above)
Comment (karenmillermoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/core/DocumentResource.kt:105]: good idea

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/core/DocumentResourceIntTest.kt (hunk 4 above)
Comment (karenmillermoj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/core/DocumentResourceIntTest.kt:187]: will add

**Pull request merged**